<h1>ACTIVITY 1: PALMER PENGUINS DATA ANALYTICS PROJECT</h1>
<hr>

<h4>ANALYST: JC RODRIGUEZ</h4>

LIBRARIES

In [5]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [6]:
bird = pd.read_csv("datasets\\penguins.csv")

In [7]:
bird

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h4>Descriptive Analytics</h4>

In [8]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [9]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h4>FILTERING A COLUMN/DATA</h4>

In [10]:
adelie_df = bird[bird["species"] == "Adelie"]

In [11]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [12]:
adelie_df["sex"] == "Male"

0      False
1      False
2      False
3      False
4      False
       ...  
147    False
148    False
149    False
150    False
151    False
Name: sex, Length: 152, dtype: bool

In [13]:
adelie_df["body_mass_g"].mean()

np.float64(3700.662251655629)

In [14]:
bird["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [15]:
bird["island"] == "Dream"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [16]:
bird["sex"] == "male"

0       True
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341     True
342     True
343    False
Name: sex, Length: 344, dtype: bool

In [17]:
bird[(bird["island"] == "Dream") & (bird["sex"] == "male") & (bird["species"] == "Adelie")]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
31   Adelie  Dream            37.2           18.1              178.0   
33   Adelie  Dream            40.9           18.9              184.0   
35   Adelie  Dream            39.2           21.1              196.0   
36   Adelie  Dream            38.8           20.0              190.0   
39   Adelie  Dream            39.8           19.1              184.0   
41   Adelie  Dream            40.8           18.4              195.0   
43   Adelie  Dream            44.1           19.7              196.0   
45   Adelie  Dream            39.6           18.8              190.0   
46   Adelie  Dream            41.1           19.0              182.0   
49   Adelie  Dream            42.3           21.2              191.0   
85   Adelie  Dream            41.3           20.3              194.0   
86   Adelie  Dream            36.3           19.5              190.0   
88   Adelie  Dream            38.3           19.2              189.0   
91   Adelie  Dream            41.1           18.1              205.0   
93   Adelie  Dream            39.6           18.1              186.0   
95   Adelie  Dream            40.8           18.9              208.0   
97   Adelie  Dream            40.3           18.5              196.0   
99   Adelie  Dream            43.2           18.5              192.0   
133  Adelie  Dream            37.5           18.5              199.0   
135  Adelie  Dream            41.1           17.5              190.0   
137  Adelie  Dream            40.2           20.1              200.0   
139  Adelie  Dream            39.7           17.9              193.0   
141  Adelie  Dream            40.6           17.2              187.0   
143  Adelie  Dream            40.7           17.0              190.0   
145  Adelie  Dream            39.0           18.7              185.0   
146  Adelie  Dream            39.2           18.6              190.0   
149  Adelie  Dream            37.8           18.1              193.0   
151  Adelie  Dream            41.5           18.5              201.0   

     body_mass_g   sex  year  
31        3900.0  male  2007  
33        3900.0  male  2007  
35        4150.0  male  2007  
36        3950.0  male  2007  
39        4650.0  male  2007  
41        3900.0  male  2007  
43        4400.0  male  2007  
45        4600.0  male  2007  
46        3425.0  male  2007  
49        4150.0  male  2007  
85        3550.0  male  2008  
86        3800.0  male  2008  
88        3950.0  male  2008  
91        4300.0  male  2008  
93        4450.0  male  2008  
95        4300.0  male  2008  
97        4350.0  male  2008  
99        4100.0  male  2008  
133       4475.0  male  2009  
135       3900.0  male  2009  
137       3975.0  male  2009  
139       4250.0  male  2009  
141       3475.0  male  2009  
143       3725.0  male  2009  
145       3650.0  male  2009  
146       4250.0  male  2009  
149       3750.0  male  2009  
151       4000.0  male  2009

In [18]:
bird.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h4>HOW TO FILLUP NULL VALUES</h4>

In [19]:
bird["bill_length_mm"] = bird["bill_length_mm"].fillna(bird["bill_length_mm"].mean())
bird["bill_depth_mm"] = bird["bill_depth_mm"].fillna(bird["bill_depth_mm"].mean())
bird["flipper_length_mm"] = bird["flipper_length_mm"].fillna(bird["flipper_length_mm"].mean())
bird["sex"] = bird["sex"].fillna(bird["sex"].mode()[0])

In [20]:
bird.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          2
sex                  0
year                 0
dtype: int64

<h4>CHANGING DATA TYPES</h4>

In [21]:
bird["species"] = bird["species"].astype('category')
bird["island"] = bird["island"].astype('category')
bird["sex"] = bird["sex"].astype('category')
bird["year"] = bird["year"].astype('category')

In [22]:
bird.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        342 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h3>Which species has a big body mass</h3>

In [23]:
bird["body_mass_g"].mean()

np.float64(4201.754385964912)

In [24]:
adelie_species = bird[bird['species'] == "Adelie"]

In [25]:
adelie_species["body_mass_g"].mean()

np.float64(3700.662251655629)

In [26]:
bird["species"].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h3>1.what is the average flippers size of male penguin of dream island?
</h3>

In [27]:
bird_dream = bird[(bird["island"] == "Dream") & (bird["sex"] == "male")]

In [28]:
bird_dream["flipper_length_mm"].mean()

np.float64(196.03174603174602)

<h3>Insights:</h3>
<h4>The average flipper size of male penguin of dream island is 196.03174603174602</h4>

<h3>2.How many male adelie does Torgersen have?</h3>

In [29]:
f = bird[(bird['species'] == "Adelie") & (bird["island"] == "Torgersen") & (bird["sex"] == "male")]

In [30]:
f

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen        39.10000       18.70000         181.000000   
3    Adelie  Torgersen        43.92193       17.15117         200.915205   
5    Adelie  Torgersen        39.30000       20.60000         190.000000   
7    Adelie  Torgersen        39.20000       19.60000         195.000000   
8    Adelie  Torgersen        34.10000       18.10000         193.000000   
9    Adelie  Torgersen        42.00000       20.20000         190.000000   
10   Adelie  Torgersen        37.80000       17.10000         186.000000   
11   Adelie  Torgersen        37.80000       17.30000         180.000000   
13   Adelie  Torgersen        38.60000       21.20000         191.000000   
14   Adelie  Torgersen        34.60000       21.10000         198.000000   
17   Adelie  Torgersen        42.50000       20.70000         197.000000   
19   Adelie  Torgersen        46.00000       21.50000         194.000000   
69   Adelie  Torgersen        41.80000       19.40000         198.000000   
71   Adelie  Torgersen        39.70000       18.40000         190.000000   
73   Adelie  Torgersen        45.80000       18.90000         197.000000   
75   Adelie  Torgersen        42.80000       18.50000         195.000000   
77   Adelie  Torgersen        37.20000       19.40000         184.000000   
79   Adelie  Torgersen        42.10000       19.10000         195.000000   
81   Adelie  Torgersen        42.90000       17.60000         196.000000   
83   Adelie  Torgersen        35.10000       19.40000         193.000000   
117  Adelie  Torgersen        37.30000       20.50000         199.000000   
119  Adelie  Torgersen        41.10000       18.60000         189.000000   
121  Adelie  Torgersen        37.70000       19.80000         198.000000   
123  Adelie  Torgersen        41.40000       18.50000         202.000000   
125  Adelie  Torgersen        40.60000       19.00000         199.000000   
127  Adelie  Torgersen        41.50000       18.30000         195.000000   
129  Adelie  Torgersen        44.10000       18.00000         210.000000   
131  Adelie  Torgersen        43.10000       19.20000         197.000000   

     body_mass_g   sex  year  
0         3750.0  male  2007  
3            NaN  male  2007  
5         3650.0  male  2007  
7         4675.0  male  2007  
8         3475.0  male  2007  
9         4250.0  male  2007  
10        3300.0  male  2007  
11        3700.0  male  2007  
13        3800.0  male  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
69        4450.0  male  2008  
71        3900.0  male  2008  
73        4150.0  male  2008  
75        4250.0  male  2008  
77        3900.0  male  2008  
79        4000.0  male  2008  
81        4700.0  male  2008  
83        4200.0  male  2008  
117       3775.0  male  2009  
119       3325.0  male  2009  
121       3500.0  male  2009  
123       3875.0  male  2009  
125       4000.0  male  2009  
127       4300.0  male  2009  
129       4000.0  male  2009  
131       3500.0  male  2009

In [31]:
f.tail(1)

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
131  Adelie  Torgersen            43.1           19.2              197.0   

     body_mass_g   sex  year  
131       3500.0  male  2009

<h3>Insights:</h3>
<h4>131 is the number of adelie male penguins on torgersen</h4>

<h3>3.What year did the female penguins get heavier?
</h3>

In [32]:
bird.groupby(["year","sex"])["body_mass_g"].mean()

year  sex   
2007  female    3821.078431
      male      4391.379310
2008  female    3887.500000
      male      4632.758621
2009  female    3874.137931
      male      4529.918033
Name: body_mass_g, dtype: float64

<h3>Insights</h3>
<h4>2008 is the year that female penguins get heavier</h4>

In [33]:
avg_mass = bird.groupby(["species", "sex"])["body_mass_g"].mean()
heaviest_combo = avg_mass.idxmax()
heaviest_value = avg_mass.max()

<h3>4.Which penguin species and sex combination has the highest average body mass, and what is that value?</h3>

In [40]:
heaviest_combo 

('Gentoo', 'male')

In [37]:
heaviest_value

np.float64(5429.615384615385)


<h3>Insights:</h3>
<h4>The species of penguin that has the highest average body mass is Gentoo Male with the value of 5429.615384615385</h4>

<h3>5.Which island has the biggest difference between the average flipper length of male and female penguins?</h3>

In [43]:
avg_flipper = bird.groupby(["island", "sex"])["flipper_length_mm"].mean()


In [44]:
avg_flipper

island     sex   
Biscoe     female    205.687500
           male      213.260400
Dream      female    190.016393
           male      196.031746
Torgersen  female    188.291667
           male      194.032686
Name: flipper_length_mm, dtype: float64


<h3>Insights:</h3>
<h4>The island that has the biggest difference between the average flipper length of male and female is Biscoe Island</h4>

<h3>6.Which year recorded the heaviest average penguin body mass?

In [45]:
avg_mass_per_year = bird.groupby("year")["body_mass_g"].mean()


In [46]:
avg_mass_per_year

year
2007    4124.541284
2008    4266.666667
2009    4210.294118
Name: body_mass_g, dtype: float64


<h3>Insights:</h3>
<h4>2008 is the year that have record of the heaviest penguin body mass</h4>

<h3>7.Which penguin species has the longest average bill length?</h3>

In [47]:
avg_bill = bird.groupby("species")["bill_length_mm"].mean()


In [48]:
avg_bill

species
Adelie       38.825144
Chinstrap    48.833824
Gentoo       47.475983
Name: bill_length_mm, dtype: float64

<h3>Insights:</h3>
<h4>Chinstrap has the longest bill length among the other species</h4>

<h3>8.On which island do penguins have the heaviest average body mass?</h3>

In [49]:
avg_mass_island = bird.groupby("island")["body_mass_g"].mean()



In [50]:
avg_mass_island

island
Biscoe       4716.017964
Dream        3712.903226
Torgersen    3706.372549
Name: body_mass_g, dtype: float64

<h3>Insights:</h3>
<h4>Biscoe has the heaviest body mass of penguins</h4>

<h3>9.Which sex (male or female) has the higher average body mass overall?</h3>

In [51]:
avg_mass_sex = bird.groupby("sex")["body_mass_g"].mean()


In [52]:
avg_mass_sex

sex
female    3862.272727
male      4518.220339
Name: body_mass_g, dtype: float64

<h3>Insights:</h3>
<h4>Male has the highest average of body mass in overall</h4>

<h3>10.Which species has the shortest average flipper length?</h3>

In [54]:
avg_flipper_species = bird.groupby("species")["flipper_length_mm"].mean()


In [55]:
avg_flipper_species


species
Adelie       190.025758
Chinstrap    195.823529
Gentoo       217.055768
Name: flipper_length_mm, dtype: float64

<h3>Insights:</h3>
<h4>Adelie species is has the shortest average of flipper lenght</h4>

<h3>11.Which island has the longest average bill length?</h3>

In [56]:
avg_bill_island = bird.groupby("island")["bill_length_mm"].mean()


In [57]:
avg_bill_island

island
Biscoe       45.249535
Dream        44.167742
Torgersen    39.046576
Name: bill_length_mm, dtype: float64

<h3>Insights:</h3>
<h4>Biscoe has the highest average of bill length</h4>

<h3>12.In which year were the penguins the lightest on average (body mass)?</h3>

In [58]:
avg_mass_year = bird.groupby("year")["body_mass_g"].mean()


In [59]:
avg_mass_year

year
2007    4124.541284
2008    4266.666667
2009    4210.294118
Name: body_mass_g, dtype: float64

<h3>Insights:</h3>
<h4>2007 is the year that penguins has the lighest average of body mass</h4>